Fix indentation, added quality of life, used current QBCore Updates, added a config file.

MatthewRorke opened the pull request:
**Describe Pull request**
new: config.lua

refactor(client-main):
- Hints use qb-core export.
- Cops stop, get out and shoot when player is near.
- Alert Police when CIT starts shooting at player.
- Move all modifiable variables to config.

fix(client-main):
- Only spawn vehicle within 200 of player location.
- Fix Indentation
- Prevent all wait outside of parent loop.

refactor(client-server):

fix(client-server):
- Indentation correction.
- Spelling errors.

If your PR is to fix an issue mention that issue here

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? **Yes**
- Does your code fit the style guidelines? **Yes**
- Does your PR fit the contribution guidelines? **Yes**

MatthewRorke committed:
new: config.lua

refactor(client-main):
- Hints use qb-core export.
- Cops stop, get out and shoot when player is near.
- Alert Police when CIT starts shooting at player.
- Move all modifiable variables to config.

fix(client-main):
- Only spawn vehicle within 200 of player location.
- Fix Indentation
- Prevent all wait outside of parent loop.

refactor(client-server):

fix(client-server):
- Indentation correction.
- Spelling errors.

MatthewRorke committed:
fix(main-client): Indentation, remove actual tabs.

MatthewRorke committed:
fix(client-main): Use self contained ped spawning.

MatthewRorke committed:
fix(main-client):
- Removed unneeded variables.
- Optimise loop.
- Indentation fix.
- Set 3rd ped into vehicle and give weapon.
- Actually use config weapons.
- hide last hint better.

MatthewRorke committed:
fix(conflicts): resolve conflicts with latest version

MatthewRorke committed:
fix(error): fix syntax error during conflict resolution

MatthewRorke committed:
fix(linting): fix linting errors

MatthewRorke committed:
fix(native): Invoke Native to be replaced by GetStreetNameAtCoord

BerkieBb commented on client/main.lua:
```diff
@@ -98,330 +84,357 @@ function AlertPolice()
-
-    end
+RegisterNetEvent('AttackTransport:InfoForLspd')
+AddEventHandler('AttackTransport:InfoForLspd', function(x, y, z)
```
You can remove AddEventHandler and pass the event handler directly to RegisterNetEvent

BerkieBb commented on client/main.lua:
```diff
@@ -431,28 +444,27 @@ function TakingMoney()
-    TriggerEvent('AttackTransport:CleanUp')
-    Citizen.Wait(2500)
+	local PedCoords = GetEntityCoords(PlayerPedId())
+	bag = CreateObject(GetHashKey('prop_cs_heist_bag_02'),PedCoords.x, PedCoords.y,PedCoords.z, true, true, true)
```
Use backticks instead of GetHashKey

BerkieBb commented on client/main.lua:
```diff
@@ -431,28 +444,27 @@ function TakingMoney()
+			break
+		end
+		hintToDisplay('Hold [G] to bail out')
+		Citizen.Wait(1)
```
Random wait 1

BerkieBb commented on fxmanifest.lua:
```diff
@@ -6,5 +6,6 @@ version '1.0.0'
 
 server_script 'server/main.lua'
 client_script 'client/main.lua'
+shared_script 'config.lua'
```
You should load the config before the ones using it

BerkieBb commented on server/main.lua:
```diff
@@ -23,27 +18,30 @@ RegisterServerEvent('AttackTransport:akceptujto', function()
 
-RegisterServerEvent('qb-armoredtruckheist:server:callCops', function(streetLabel, coords)
+RegisterServerEvent('qb-armoredtruckheist:server:callCops')
+AddEventHandler('qb-armoredtruckheist:server:callCops', function(streetLabel, coords)
```
You added the event handler? There is no need for it

MatthewRorke committed:
fix(hash): use backtick syntax for hashes.
fix(config): put shared script before scripts that use them.
fix(events): register events with callback argument instead of AddEventHandler.
fix(wait): Wait(0) instead of 1.

BerkieBb commented on client/main.lua:
```diff
@@ -40,53 +31,48 @@ RegisterNetEvent('QBCore:Client:OnJobUpdate', function(JobInfo)
 end)
 
 function hintToDisplay(text)
```
This function can be local

BerkieBb commented on client/main.lua:
```diff
@@ -40,53 +31,48 @@ RegisterNetEvent('QBCore:Client:OnJobUpdate', function(JobInfo)
+	exports['qb-core']:DrawText(text)
+end
+
+function hideLastHint()
```
This function can be local

BerkieBb commented on client/main.lua:
```diff
@@ -40,53 +31,48 @@ RegisterNetEvent('QBCore:Client:OnJobUpdate', function(JobInfo)
+					Citizen.Wait(500)
+				end
+			end
+		else
```
Instead of this else statement, add a sleep variable and change the sleep to 0 when the dist is under 25.0 but make the default value of sleep 500 so it changes like that

BerkieBb commented on client/main.lua:
```diff
@@ -40,53 +31,48 @@ RegisterNetEvent('QBCore:Client:OnJobUpdate', function(JobInfo)
+				TaskStartScenarioInPlace(dealer, "WORLD_HUMAN_AA_SMOKE", 0, false)
+			end
+			if dist <= 2.0 then
+				QBCore.Functions.DrawText3D(Config.MissionMarker.x, Config.MissionMarker.y, Config.MissionMarker.z, "~b~[E]~w~ To accept mission")
```
Never use this function, replace it by qb-core drawtextui

BerkieBb commented on client/main.lua:
```diff
@@ -40,53 +31,48 @@ RegisterNetEvent('QBCore:Client:OnJobUpdate', function(JobInfo)
 end)
 ---
 
 function CheckGuards()
```
This one can be local as well

BerkieBb commented on client/main.lua:
```diff
@@ -40,53 +31,48 @@ RegisterNetEvent('QBCore:Client:OnJobUpdate', function(JobInfo)
+	end
 end
 
 function AlertPolice()
```
Every function can be localised here actually

BerkieBb commented on client/main.lua:
```diff
@@ -99,329 +85,350 @@ function AlertPolice()
+				TaskPlayAnim(PlayerPedId(), dict, "fob_click_fp", 8.0, 8.0, -1, 48, 1, false, false, false)
+				TriggerEvent('AttackTransport:CleanUp')
+				RemoveBlip(TruckBlip)
+				Citizen.Wait(500)
```
This is not needed, you're not in a loop and the wait is useless

BerkieBb commented on client/main.lua:
```diff
@@ -99,329 +85,350 @@ function AlertPolice()
+				Citizen.Wait(500)
+			end
+		end
+
```
Random empty space

BerkieBb commented on client/main.lua:
```diff
@@ -99,329 +85,350 @@ function AlertPolice()
+					end
+				end
+			end
+			Wait(0)
```
This one can have a sleep of 500 as well before turning to 0 after the dist is under 55

BerkieBb commented on client/main.lua:
```diff
@@ -99,329 +85,350 @@ function AlertPolice()
+		else
+			Wait(1500)
+		end
+		Wait(0)
```
Use sleep here as well

BerkieBb commented on server/main.lua:
```diff
@@ -66,7 +63,7 @@ RegisterServerEvent('AttackTransport:graczZrobilnapad', function()
 
 	if chance >= 95 then
-		xPlayer.Functions.AddItem('security_card_01', 1)
+		xPlayer.Functions.AddItem('security_card_00', 1)
```
Change this back, we don't have a `00` security card

MatthewRorke committed:
fix(scope): Make functions used within client-only local.
fix(scope): DrawText3D to be local and not used from core.
fix(loops): Use SLEEP variable instead of multiple Citizen.Wait.
fix(space): Remove random space.

MatthewRorke committed:
fix(scope): Remove Citizen. in order to use GlobalScope for optimised runtime.

Z3rio committed:
Merge branch 'main' into main

sinphoenix reviewed (approved):
Tested!!

GhzGarage committed:
Merge branch 'main' into main

GhzGarage committed:
Merge branch 'main' into main
